eclass/rpm-extended.eclass

```diff
@@ -0,0 +1,76 @@
+# Copyright 1999-2020 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+# Author : Andrew Ammerlaan <andrewammerlaan@riseup.net> (16-12-2020)
+#
+# Extension of the rpm.eclass by Alastair Tse <liquidx@gentoo.org>
+#
+# This also installs *all* files in the rpm, and extracts
+# post/pre(un)install scripts and runs them.
+#
+# This is mostly useful when creating ebuilds for rpm files in bulk.
+# In cases when it is not doable to write a Gentoo specific src_install
+# and pkg_post/pre/install/rm functions for each ebuild.
+
+inherit rpm
+
+# Need rpm to extact scripts
+BDEPEND="app-arch/rpm"
+
+# Otherwise we get the S does not exist error
+S="${WORKDIR}"
+# Everything in the rpm is prebuilt
+QA_PREBUILT="*"
+
+rpm-extended_src_compile() {
```

Comment:
maybe add a very small documentation header saying it does nothing like: https://gitweb.gentoo.org/repo/gentoo.git/tree/eclass/python-single-r1.eclass#n519